File: LeetCode/Math/JS/Leetcode1447-最简分数.js
var simplifiedFractions = function(n) {
    let gcd = function(a,b) {
        return b ? gcd(b,a % b) : a;
    }
    const ans = [];
    for(let i = 2;i <= n;i++) {
        for(let j = 1;j < i;j++) {
            if(gcd(i,j) == 1) ans.push(j + '/' + i);
        }
    }
    return ans;
};

File: LeetCode/Math/Golang/Leetcode1447-最简分数.go
package sbgo

import "strconv"

func simplifiedFractions(n int) []string {
	var gcd func(a, b int) int
	gcd = func(a, b int) int {
		if b == 0 {
			return a
		}
		return gcd(b, a%b)
	}
	ans := []string{}
	for i := 2; i <= n; i++ {
		for j := 1; j < i; j++ {
			if gcd(i, j) == 1 {
				ans = append(ans, strconv.Itoa(j)+"/"+strconv.Itoa(i))
			}
		}
	}
	return ans
}
